examples/vision/depth_estimation.py

```diff
@@ -353,28 +358,28 @@ def __init__(self):
 
     def calculate_loss(self, target, pred):
```

Comment:
You can remove the custom `train_step` and instead override `compute_loss()` (it would be similar to this method). This should enable the code to run with all backends.

Reply:
Removed `train_step` and `test_step` as well by overriding `compute_loss()`. Attaching tested [gist](https://colab.sandbox.google.com/gist/SuryanarayanaY/b0cec18a72b6f3bbe12235da72eb3b55/depth_estimation_keras3_testing_r2.ipynb) for reference.